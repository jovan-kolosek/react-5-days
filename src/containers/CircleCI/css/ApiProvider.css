.api_provider-container {
  position: fixed;
  top: 35%;
  left: 50%;
  transform: translate(-50%, -50%);
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -o-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
}

.api_provider-form {
  width: 500px;
  margin-top: 20px;
}

.api_provider-form__input {
  width: 350px;
  height: 30px;
  margin-right: 20px;
  padding-left: 10px;
}

.api_provider-form__button {
  width: 60px;
  height: 36px;
}

.api_provider-form__error {
  color: red;
  margin-top: 12px;
}
