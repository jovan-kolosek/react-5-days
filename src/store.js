import { applyMiddleware, createStore, compose } from 'redux';
import rootReducer from './reducers';
import thunk from 'redux-thunk';
import { logger } from './middleware';

const middleware = [thunk, logger];

const store = compose(applyMiddleware(...middleware))(createStore)(rootReducer);

export default store;
