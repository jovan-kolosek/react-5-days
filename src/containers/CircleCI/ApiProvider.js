import React, {Component} from 'react';
import { Redirect } from 'react-router';

import './css/ApiProvider.css'

class ApiProvider extends Component {

  state = { 
    input: '',
    redirect: false, 
  }

  handleSave = () => {
    sessionStorage.setItem('CIRCLE_CI_API_KEY', this.state.input)
    this.setState({ redirect: true })
  }

  handleOnChange = (event) => {
    this.setState({ input: event.target.value })
  }

  render() {
    if(this.state.redirect) return <Redirect to='/' />
    return (
      <div className="api_provider-container">
        <form className="api_provider-form">
          <input className="api_provider-form__input" onChange={this.handleOnChange} placeholder='Please add CircleCI API key' />
          <button className="api_provider-form__button" onClick={this.handleSave}>Save</button>
        </form>
      </div>
    );
  }
}

export default ApiProvider;
