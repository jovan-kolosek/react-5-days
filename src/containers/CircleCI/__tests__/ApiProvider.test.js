import React from 'react';
import ApiProvider from '../ApiProvider';
import { Provider } from 'react-redux';
import store from '../../../store';
import renderer from 'react-test-renderer';

test('ApiProvider with default context renders', () => {
  const component = renderer.create(
    <Provider store={store}><ApiProvider /></Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});