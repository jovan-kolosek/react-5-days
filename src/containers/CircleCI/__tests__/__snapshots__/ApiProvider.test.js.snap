// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiProvider with default context renders 1`] = `
<div
  className="api_provider-container"
>
  <form
    className="api_provider-form"
  >
    <input
      className="api_provider-form__input"
      onChange={[Function]}
      placeholder="Please add CircleCI API key"
    />
    <button
      className="api_provider-form__button"
      onClick={[Function]}
    >
      Save
    </button>
    <div
      className="api_provider-form__error"
    >
      
    </div>
  </form>
</div>
`;
